File: python-code/ex7/professor.py
class Professor:
  def __init__(self, nome):
    self.nome = nome
    self.escolas = []

  def adicionar_escola(self, escola):
    if escola not in self.escolas:
      self.escolas.append(escola)
      escola.adicionar_professor(self)

  def __str__(self):
    return self.nome

File: java-code/src/main/java/ex7/Professor.java
package ex7;

import java.util.ArrayList;
import java.util.List;

public class Professor {
    private String nome;
    private List<Escola> escolas;

    public Professor(String nome) {
        this.nome = nome;
        this.escolas = new ArrayList<>();
    }

    public void adicionarEscola(Escola escola) {
        if (!escolas.contains(escola)) {
            escolas.add(escola);
            escola.adicionarProfessor(this);
        }
    }

    public List<Escola> getEscolas() {
        return escolas;
    }

    @Override
    public String toString() {
        return nome;
    }
}
